Libraries/LibWebView/WebContentClient.cpp

```diff
@@ -470,24 +471,25 @@ void WebContentClient::did_change_favicon(u64 page_id, Gfx::ShareableBitmap favi
     }
 }
 
-Messages::WebContentClient::DidRequestAllCookiesResponse WebContentClient::did_request_all_cookies(URL::URL url)
+void WebContentClient::did_request_all_cookies(URL::URL url, DidRequestAllCookies::Resolver resolve)
 {
-    return Application::cookie_jar().get_all_cookies(url);
+    resolve(Application::cookie_jar().get_all_cookies(url));
 }
 
-Messages::WebContentClient::DidRequestNamedCookieResponse WebContentClient::did_request_named_cookie(URL::URL url, String name)
+void WebContentClient::did_request_named_cookie(URL::URL url, String name, DidRequestNamedCookie::Resolver resolve)
 {
-    return Application::cookie_jar().get_named_cookie(url, name);
+    resolve(Application::cookie_jar().get_named_cookie(url, name));
 }
 
-Messages::WebContentClient::DidRequestCookieResponse WebContentClient::did_request_cookie(URL::URL url, Web::Cookie::Source source)
+void WebContentClient::did_request_cookie(URL::URL url, Web::Cookie::Source source, DidRequestCookie::Resolver resolve)
 {
-    return Application::cookie_jar().get_cookie(url, source);
+    resolve(Application::cookie_jar().get_cookie(url, source));
 }
 
-void WebContentClient::did_set_cookie(URL::URL url, Web::Cookie::ParsedCookie cookie, Web::Cookie::Source source)
+void WebContentClient::did_set_cookie(URL::URL url, Web::Cookie::ParsedCookie cookie, Web::Cookie::Source source, DidSetCookie::Resolver resolve)
 {
     Application::cookie_jar().set_cookie(url, cookie, source);
+    return resolve();
```

Comment:
`return`? This is a `void` function

Reply:
(SideNote: There is one case where a "valued" void return makes sense, and that is enforcing tail calls with compiler attributes)

Reply:
> (SideNote: There is one case where a "valued" void return makes sense, and that is enforcing tail calls with compiler attributes)

Thank you for your input. The parts involving actual IPC message implementations are just to show how the new system would work - and so I did a sort of transpiling of the implementations, I myself did not know or understand why "returning" a void was needed.

Reply:
> `return`? This is a `void` function

This came as a part of a re-write of old implementations to show how it should be used. In this particular case doing the first scenario below works just fine, since there's no other branching etc going on after that. But both scenarios are useful:

```cpp
if(foo) {
  resolve();
  return;
}
```

can be spelled

```cpp
// When resolving, the function should also return
if(foo)
  return resolve();
```

Since the callbacks themselves are guaranteed to return void, this is fine to do.

Reply:
Btw, @ADKaster you wrote IPC messages that return void responses, see `UIProcessServer.ipc`: whether the resolve mechanism is promise or a `Resolver` an empty resolve needs to happen.

If what you wrote there is a bug, it should probably be fixed separately from this patch series, right?